packages/starlight/index.ts

```diff
@@ -15,11 +15,20 @@ import {
   StarlightConfig,
   StarlightConfigSchema,
 } from './utils/user-config';
+import { errorMap } from './utils/error-map';
 
 export default function StarlightIntegration(
   opts: StarlightUserConfig
 ): AstroIntegration[] {
-  const userConfig = StarlightConfigSchema.parse(opts);
+  const parsedConfig = StarlightConfigSchema.safeParse(opts, {
+    errorMap
+  });
+
+  if (!parsedConfig.success) {
+    throw new Error(parsedConfig.error.issues.map(i => i.message).join('\n'));
```

Comment:
What do you think of prepending a line explaining that the error is in the `starlight` integration config?

Based on the example you shared, I’m imagining something like

```
15:23:05 [astro] Unable to load /home/otterlord/Documents/Repos/starlight/docs/astro.config.mjs

 error   Invalid config passed to starlight integration
  **logo**: Did not match union:
  > **logo.replacesTitle**: Expected type `"boolean"`, received "string"
```

Reply:
Yeah, that sounds good :+1: